fitapp/tasks.py

```diff
@@ -46,24 +46,22 @@ def unsubscribe(*args, **kwargs):
         raise Reject(exc, requeue=False)
 
 
-
 @shared_task
-def get_time_series_data(fitbit_user, cat, resource, date=None):
+def get_time_series_data(fitbit_user, categories=[], date=None):
     """ Get the user's time series data """
 
-    try:
-        _type = TimeSeriesDataType.objects.get(category=cat, resource=resource)
-    except TimeSeriesDataType.DoesNotExist:
-        logger.exception("The resource %s in category %s doesn't exist" % (
-            resource, cat))
+    filters = {'category__in': categories} if categories else {}
+    types = TimeSeriesDataType.objects.filter(**filters)
+    if not types.exists():
+        logger.exception("Couldn't find the time series data types")
         raise Reject(sys.exc_info()[1], requeue=False)
 
     # Create a lock so we don't try to run the same task multiple times
     sdat = date.strftime('%Y-%m-%d') if date else 'ALL'
-    lock_id = '{0}-lock-{1}-{2}-{3}'.format(__name__, fitbit_user, _type, sdat)
+    cats = '-'.join('%s' % i for i in categories)
+    lock_id = '{0}-lock-{1}-{2}-{3}'.format(__name__, fitbit_user, cats, sdat)
     if not cache.add(lock_id, 'true', LOCK_EXPIRE):
```

Comment:
@brad I think the easiest solution is to punt on it for now and make a note in the README here that the fitbit tasks shouldn't be run concurrently, and then give an example of a way to set up celery to do that. I think we can use celery's [manual routing](http://docs.celeryproject.org/en/latest/userguide/routing.html#manual-routing) feature to create a new queue and then when starting celery, make sure there is only one thread working on that queue.

It would be much nicer to support concurrent tasks (but trickier too). I think we can do the locking with the db. One idea is to store the lock in the DB, use the `@transaction.atomic()` decorator like you said, and Django's [select_for_update](https://docs.djangoproject.com/en/1.8/ref/models/querysets/#select-for-update) to acquire the lock. I think it would be enough to have one lock per user so that only tasks for one user can execute at a time. That way we shouldn't have multiple processes trying to renew the token at the same time and stepping on each other.